prime-router/docs/design/proposals/translation-template-test-plan.md

```diff
@@ -0,0 +1,47 @@
+# Translation Template Test Plan
+
+## Problem Statement
+As a Reportstream engineer, I want tests that verify that the changes I make to translations will not break essential 
+rules set forth by the spec or our users. 
+
+Currently, we use integration tests for this purpose, but these do not give us the ability to easily tell which thing 
+broke/changed or even really what is specifically being tested.
+
+## Considerations
+- We not only want to check the rules set forth in the spec, but the ones set forth by our users.
+
+## Proposal on *what* to test
+We should write tests when:
+- The spec specifies a condition
+- There is a requirement specified by a sender or receiver. This means that there needs to be a test for every rule in
+  a sender or receiver transform, This includes the default transform since these are ReportStream specific rules we
+  are implementing
+- We have created an extension not specified in the spec
+
+## Options for *how* to test 
+1. Continue with integration tests
+    - These would specific to the field they are trying to test rather than a cluster of fields
+    - Named after the field they are trying to test
+
+   - Pros:
+     - Setup already exists, works, and it tested 
+   - Cons:
+     - The file to add/updates/remove tests would become difficult to read and find tests in because of the sheer number
+     - Many files would need to be added
+2. Unit Tests (Proposal)
```

Comment:
I think unit tests are a great fit for testing out specific segments/resources

Reply:
Got it, so your vote is unit tests? @jack-h-wang was voting for integration tests with cleanup to make them nicer. Not really sure what call to make here at this point. Maybe we can do a 17th on this tomorrow after standup?